scene/animation/tween.cpp

```diff
@@ -787,11 +787,13 @@ bool Tween::start() {
 	if (pending_update != 0) {
 		// Start the tweens after deferring
 		call_deferred("start");
+		ERR_FAIL_COND_V_MSG(!is_inside_tree(), false, "Tween was not added to the SceneTree!");
 		return true;
 	}
 
 	// We want to be activated
 	set_active(true);
+	ERR_FAIL_COND_V_MSG(!is_inside_tree(), false, "Tween was not added to the SceneTree!");
```

Comment:
Only one occurrence should be needed at the start of the method, no? Here it will still set it active, even though it raises an error and reports `false`.

Reply:
It depends on the expected/wanted behaviour 🤔 

I implemented it as follows:
- Tween that is not in SceneTree is started (Tween.start()) and raises an error.
- Tween enters the SceneTree and is started using the ENTER_TREE signal/connection

![Capture](https://user-images.githubusercontent.com/42484461/64075633-04068e80-ccbb-11e9-9f4e-4b9ff2716664.PNG)

Would it be better to not make the active the Tween active and just throw the error?

Reply:
IMO an error means that the call fails, and the user should fix their code so that they don't generate an error. So I'd make it a proper failure and users should take care to call `start()` when in the scene tree.

The way you implemented it, it's more a warning that tells them that the call was wrong, but has been handled gracefully nevertheless so that it works when added to the tree.

Reply:
Ok, makes sense! I changed it!

Reply:
Did you forget to push though? :)

Reply:
Gimme a sec, I'm trying to squash my commits and force-push them without using Gitkraken for the first time ever.

Reply:
Should be fine now though 😉